src/common/simulation.ts

```diff
@@ -44,16 +47,58 @@ export default abstract class Simulation {
     this._stepCounter = 0;
     this._enemyIdCounter = enemyIdCounter;
     this._gameMap = map;
+    this._wave = new Wave(
+      1,
+      constants.WAVE_ENEMY_COUNT_INCREMENT,
+      constants.WAVE_ENEMY_HEALTH_INCREMENT,
+    );
+    this._waveClearStep = 0;
   }
 
   public commonUpdate(): void {
     this._stepCounter += 1;
 
-    //spawns a baby yoda per second
-    if (this._stepCounter % Math.floor(1 / this.updateStep) === 0) {
-      this.addEnemy();
+    const killed = this.despawnEnemies();
+    this._wave.kill(killed);
+
+    if (this._wave.finished) {
+      // mark this instant as when the wave was finished
+      if (this._waveClearStep === 0) {
+        this._waveClearStep = this._stepCounter - 1;
+        // TODO: remove this when we display the wave information on screen
+        console.info(`Finished wave ${this._wave.waveNumber}`);
+      }
+
+      // create a new wave
+      if (
+        (this._stepCounter - this._waveClearStep) %
+          (constants.WAVE_COOLDOWN * constants.SERVER_UPS) ===
+        0
```

Comment:
```suggestion
        (this._stepCounter - this._waveClearStep) >=
          (constants.WAVE_COOLDOWN * constants.SERVER_UPS)
```
Vill vi inte bara kolla när det har gått tillräckligt med steps en gång?